dts/bindings/sensor/ti,tmp435.yaml

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2024 Bittium Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: Texas Instruments TMP435 temperature sensor
+
+compatible: "ti,tmp435"
+
+bus: tmp435
+
+include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
+  external-channel:
+    description: External temperature measurement is enabled
+    type: boolean
+  resistance-correction:
+    description: Resistance correction feature for the external
+                 temperature channel. If not enabled then conversion
+                 is faster but with lower accuracy.
+    type: boolean
+  beta-compensation:
+    description: Beta Compensation Configuration, see Table 8
+                 from TMP435 data sheet, some common values.
+                 0x0f, Automatically selected range 7 (beta > 27.0) or
+                       Automatically detected diode connected sensor
+                 0x07, Manually disabled beta correction
+    type: int
+    default: 0x0f
```

Comment:
Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

Reply:
Done